index.d.ts

```diff
@@ -1,49 +1,68 @@
-import * as fastify from "fastify";
-import { IncomingMessage, Server, ServerResponse } from "http";
-import { DecodeOptions, Secret, SignOptions, VerifyCallback, VerifyOptions } from "jsonwebtoken";
-
-declare module "fastify"
-{
-    interface Jwt
-    {
-        decode: (token: string, options?: DecodeOptions) => null | { [key: string]: any } | string;
-        options: {
-            /**
-             * decodeOptions
-             */
-            decode: DecodeOptions,
-
-            /**
-             * signOptions
-             */
-            sign: SignOptions,
-
-            /**
-             * verifyOptions
-             */
-            verify: VerifyOptions,
-        };
-        secret: Secret;
-        sign: (playload: string | Buffer | object, options?: SignOptions, callback?: VerifyCallback) => string;
-        verify: (token: string, options?: VerifyOptions, callback?: VerifyCallback) => Promise<object | string>;
-    }
-    interface FastifyInstance<HttpServer = Server, HttpRequest = IncomingMessage, HttpResponse = ServerResponse>
-    {
-        jwt: Jwt;
-    }
+import * as fastify from 'fastify';
+import * as http from 'http';
+import * as jwt from 'jsonwebtoken';
+
+declare module 'fastify' {
+  namespace JWTTypes {
+    type PayloadType = object | string;

```

Comment:
Update this to the following intersection: `string | object | Buffer | NodeJS.ReadableStream`
Then you can remove the `| Buffer` from the `sign` method parameters

Reply:
`jsonwebtoken` library's `sign()` allow to pass `object` or ` Buffer` or `string` (See [here](https://github.com/auth0/node-jsonwebtoken#jwtsignpayload-secretorprivatekey-options-callback)).

However, `jsonwebtoken` library's `verify()` and `decode()` returns `string` or `object`, because these use `jwt` library's `decode()` (See [here](https://github.com/brianloveswords/node-jws/blob/ab660dbf716bfbff4eda3f9cf6e318cf7e3bfb29/lib/verify-stream.js#L57-L78)).

Reply:
Then in this case I'd rather see multiple type definitions such as `SignPayloadType` and `VerifyPayloadType` etc.

Reply:
Ok, I'll rewrite.